ts/parsers/paragraphparser.ts

```diff
@@ -1,37 +1,126 @@
+        };
+    }
+
+    public static getListlevel(paragraph): number {
+        const level = getValueAtPath(paragraph, '["a:pPr"][0]["$"]["lvl"]');
+
+        return level ? parseInt(level) : 0;
+    }
+
+    public static getListType(paragraph): ListType {
+        if (checkPath(paragraph, '["a:pPr"][0]["a:buAutoNum"]')) {
+            return ListType.Ordered;
+        } else { //getValueAtPath(paragraph, '["a:pPr"][0]["a:buChar"]')
+            return ListType.UnOrdered;
+        }
+    }
+
     public static extractParagraphElements(paragraphs: any[]): PowerpointElement["paragraph"] {
         if (!paragraphs || paragraphs.length === 0) {
             return null;
         }
 
-        return paragraphs.map((paragraph) => {
-            const textElements = paragraph["a:r"] || [];
-            const content = textElements.map((txtElement) => {
-                return {
-                    text: txtElement["a:t"] || "",
-                    textCharacterProperties: this.determineTextProperties(getValueAtPath(txtElement, '["a:rPr"][0]'))
-                };
-            });
+        const paras = [];
+        const stack = [];
+        const paragraph: Paragraph = {
+            list: {
+                listType: ListType.Ordered,
+                listItems: []
+            }
+        };
+        let currentParagraph = paragraph;
+        let currentLevel = -1;
 
-            return {
-                content: content,
-                paragraphProperties: this.determineParagraphProperties(paragraph)
-            };
-        });
+        for (const p of paragraphs) {
+            if (this.isList(p)) {
+                const listLevel = this.getListlevel(p);
+
+                //if its the first of the list kind
+                if (currentLevel === -1) {
+                    currentLevel = 0;
+                    currentParagraph.list.listType = this.getListType(p);
+                    currentParagraph.list.listItems.push(this.getParagraph(p));
+                    stack.push(currentParagraph);
+                }
+                //if the level is same keep pushing the list items in the same array
+                else if (listLevel === currentLevel) {
+                    currentParagraph.list.listItems.push(this.getParagraph(p));
+                } else if (listLevel > currentLevel) {
+                    //if there is another hierarchy starting create a new list for it
+                    const newPara: Paragraph = {
+                        list: {
+                            listType: this.getListType(p),
+                            listItems: [this.getParagraph(p)]
+                        }
+                    };
+                    currentParagraph.list.listItems.push(newPara);
+                    currentParagraph = newPara;
+                    //pushing it in the stack to keep track of the parents
+                    stack.push(newPara);
+                    currentLevel++;
```

Comment:
![image](https://user-images.githubusercontent.com/24408637/125578350-af9989db-90c4-462a-9f14-df30e3ed0ede.png)

Would this cater for jumps UP multiple levels? I know we discussed jumping down several levels but I wonder if the case of going up multiple levels is also considered?

Reply:
No it probably doesn't. assumed it will always go one level above. i don't see why someone would do that :grimacing:
Its a major change in the existing logic, might need to redo the whole logic.
Should we handle it now, or leave it for later ?

Reply:
That screenshot gives me very uncomfortable feelings 😄 trying to avoid looking at that, its just so wrong lol

Reply:
haha .. I wouldn't it put it beyond people to add double tab to add extra spacing .. so I think it is possible so yes, I think we do need to support it.

Having said that, I'm happy to leave that for later (but before the release if we can find time)

> That screenshot gives me very uncomfortable feelings

It was specifically designed to do exactly that :)

Reply:
Its been taken care of now. Fixed and working as expected. Need @kryasaleg to confirm if he is happy with it.  Or if he needs any further changes from me on this structure.

Reply:
thanks!